src/test/java/ibm/jceplus/junit/base/BaseTestRSAKeyInterop.java

```diff
@@ -536,7 +537,18 @@ public void testEncryptPlusDecryptJCE() {
 
             RSAPublicKey rsaPubPlus = (RSAPublicKey) rsaKeyPairPlus.getPublic();
             rsaKeyPairPlus.getPrivate();
-            Cipher cipherPlus = Cipher.getInstance("RSA/ECB/PKCS1Padding", providerName);
+            Cipher cipherPlus = null;
+            try {
+                cipherPlus = Cipher.getInstance("RSA/ECB/PKCS1Padding", providerName);
+            } catch (NoSuchAlgorithmException nsae) {
+                nsae.printStackTrace();
+                assertTrue("RSA is not instantiated in Cipher for " + providerName,
+                            (nsae instanceof NoSuchAlgorithmException && 
```

Comment:
Could we break this statement into 3 assertions all of a different types? For example we could have `assertEquals(string1, string2) which then would print the string if it does not match. Advantage here would be that it would be more precise upon failure instead of us knowing that one ( or more ) of the 3 conditions failed.

Reply:
Updated